packages/neo-one-client/src/__tests__/user/LocalUserAccountProvider.test.js

```diff
@@ -0,0 +1,1118 @@
+/* @flow */
+import BigNumber from 'bignumber.js';
+
+import { common, utils } from '@neo-one/client-core';
+
+import LocalUserAccountProvider from '../../user/LocalUserAccountProvider';
+import LocalKeyStore from '../../user/keystore/LocalKeyStore';
+import NEOONEProvider from '../../provider/neoone/NEOONEProvider';
+import * as clientUtils from '../../utils';
+import {
+  InsufficientFundsError,
+  InvalidTransactionError,
+  InvokeError,
+  NoAccountError,
+  NothingToClaimError,
+  NothingToTransferError,
+} from '../../errors';
+
+describe('LocalUserAccountProvider', () => {
+  const id1 = {
+    network: 'net1',
+    address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
+  };
+  const account1 = {
+    type: 'test',
+    id: id1,
+    name: 'name1',
+    scriptHash: '0xecc6b20d3ccac1ee9ef109af5a7cdb85706b1df9',
+    publicKey: 'publicKey1',
+  };
+  const options = {
+    from: id1,
+    attributes: [{ usage: 'Remark', data: 'testData' }],
+    networkFee: new BigNumber('1'),
+  };
+  const outputs = {
+    oneNEO: {
+      asset: common.NEO_ASSET_HASH,
+      value: new BigNumber('1'),
+      address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
+      txid:
+        '0x7f48028c38117ac9e42c8e1f6f06ae027cdbb904eaf1a0bdc30c9d81694e045c',
+      vout: 1,
+    },
+    sevenNEO: {
+      asset: common.NEO_ASSET_HASH,
+      value: new BigNumber('7'),
+      address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
+      txid:
+        '0x7f48028c38117ac9e42c8e1f6f06ae027cdbb904eaf1a0bdc30c9d81694e045d',
+      vout: 2,
+    },
+    elevenGAS: {
+      asset: common.GAS_ASSET_HASH,
+      value: new BigNumber('11'),
+      address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
+      txid:
+        '0x7f48028c38117ac9e42c8e1f6f06ae027cdbb904eaf1a0bdc30c9d81694e045e',
+      vout: 3,
+    },
+  };
+  const results = {
+    fault: {
+      state: 'FAULT',
+      gasConsumed: new BigNumber('3'),
+      stack: [{ type: 'String', value: 'val' }],
+      message: 'testMessage',
+    },
+    halt: {
+      state: 'HALT',
+      gasConsumed: utils.ZERO_BIG_NUMBER,
+      stack: [{ type: 'String', value: 'val' }],
+    },
+  };
+  const witness = {
+    invocation: '02028a99826ed',
+    verification: '02028a99826ee',
+  };
+
+  let wallets = [];
+  const store = {
+    type: 'test',
+    getWallets: () => Promise.resolve(wallets),
+    saveWallet: wallet => {
+      wallets.push(wallet);
+      return Promise.resolve();
+    },
+    deleteWallet: wallet => {
+      wallets.splice(wallets.indexOf(wallet), 1);
+      return Promise.resolve();
+    },
+  };
+
+  let keystore = new LocalKeyStore({ store });
```

Comment:
Looks all the methods called on these are being mocked out, any particular reason to instantiate a real version here? You could just use an empty object.